src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -158,6 +160,53 @@
      */
     long findLargestPresentTranslogId() throws IOException;
 
+    /**
+     * Returns an OperationIterator to iterate over all translog entries in the given translog ID.
+     * @throws java.io.FileNotFoundException if the file for the translog ID can not be found
+     */
+    OperationIterator openIterator(long translogId) throws IOException;
+
+    /**
+     * Iterator for translog operations.
+     */
+    static class OperationIterator implements Closeable {
```

Comment:
This is very close tot the snapshot API and I think they should be merged. I'm ok with leaving this as is for now, as the change I'm working on touches all of this anyway. It's a shame to do double work here.